server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/change/TypeNameFactory.scala

```
@@ -0,0 +1,16 @@
+package org.apache.james.jmap.change
+
+import javax.inject.Inject
+import scala.jdk.CollectionConverters._
+
+case class TypeNameFactory @Inject()(setTypeName: java.util.Set[TypeName]) {
+  val all: scala.collection.mutable.Set[TypeName] = setTypeName.asScala
+
+  def parse(string: String): Either[IllegalArgumentException, TypeName] = {
+    all.foreach(typeName => typeName.parse(string).fold(
+      e => e.printStackTrace(),
```

I think we can safely discard the error here (`{}`)